src/preamble.js

```diff
@@ -621,6 +629,7 @@ function abort(what) {
   // in code paths apart from instantiation where an exception is expected
   // to be thrown when abort is called.
   throw e;
```

Comment:
I think it might make sense to delete this path and just use `__trap` in all cases.. but I guess that can be a followup maybe?

Reply:
Yeah I didn't do that in this PR because
1. This changes the error message slightly for existing users, which I don't think is a big deal, but just to be conservative
2. I eventually want to revive this, by making JS API that can create traps; so I was somewhat hesitant to delete this part.

I don't think either of these is an important enough though. WDYT?

Reply:
I'm OK with landing this now and then looking at possible simplification later.

I think I'm currently a fan of (in release builds) avoiding the JS import of abort completely.. and keeping this fulling native where possible.